Sources/Remote Logging/Crash Logging/SentryEventJSException.swift

```diff
@@ -0,0 +1,98 @@
+import Foundation
+import Sentry
+
+public struct JSException {
+    public let type: String
+    public let value: String
+    public let stacktrace: [StacktraceLine]
+    public let context: [String: Any]
+    public let tags: [String: String]
+    public let isHandled: Bool
+    public let handledBy: String
```

Comment:
Nitpick: If `handledBy` is declared as `String?`, I think the `isHandled` property can be implied as `var isHandled: Bool { handledBy != nil }`?

Reply:
Although both seem connected, I'm hesitant to infer that having `handledBy` defined implies that `isHandled` should be `false`. Let me explain a couple of cases to clarify it.

`isHandled` determines if the exception is handled by the editor, currently, this happens for exceptions captured by the error boundary components ([reference](https://github.com/WordPress/gutenberg/blob/c017e122aa58dbfcc5e33fad72a1b9207ac77880/packages/block-editor/src/components/block-list/block-crash-boundary.native.js#L20-L31)). When an error is encountered, for instance, at block level we display an error message but let the user continue editing a post. In this case, `handledBy` will tell the mechanism used to handle it and `isHandled` will be `true`.

On the other hand, other exceptions can be captured via the global error handler ([reference](https://github.com/wordpress-mobile/gutenberg-mobile/blob/f1698d65c22ce90a9552e0c6ce316c3c50d10660/src/errorLogging/index.js#L14-L43)). These are exceptions produced outside the error boundaries. Here, we use `handledBy` to tell that the exception was captured by the global error handler but we mark them as unhandled (i.e. `isHandled` as `false`) because the editor will crash.

Reply:
Thanks for the explanation!